spec/requests/courses_spec.rb

```diff
@@ -22,95 +22,154 @@
   let(:subject) { FactoryBot.create(:subject) }
   let(:valid_attributes) { { name: "Biology 101" } }
   let(:invalid_attributes) { { name: nil } }
-
-  before { login_as(user, scope: :user) }
+  let(:login) { login_as(user, scope: :user) }
```

Comment:
issue: Best not to do the log-in/sign-in in a let statement since it is not returning a value that you are using in other places. You could do it in a before block, or, even better, do it in the `it` block like you do on line 37.